src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java

```diff
@@ -0,0 +1,277 @@
+package org.jenkinsci.plugins.saml;
+
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
```

Comment:
All this classes are internal proprietary APIs and may be removed in a future release (as noted during compilation). Did you check if [bouncycastle-api-plugin](https://github.com/jenkinsci/bouncycastle-api-plugin/blob/06785880330f3c52113e5f18fbc13e9d46162de6/src/main/java/jenkins/bouncycastle/api/PEMEncodable.java) provides what you need here?

This is a :bug: IMO